test/CMakeLists.txt

```diff
@@ -64,6 +64,42 @@ if(MIOPEN_BACKEND_HIP AND NOT MIOPEN_TEST_HIP_NOGPU)
     set(MIOPEN_TEST_HIP TRUE)
 endif()
 
+# Detect GPU type for testing.
+# For HIP_NOGPU backend, GPU detection is not required and should be disabled.
+# Also we do not detect GPU when target GPU for testing is specified explicitly.
+if(NOT (MIOPEN_TEST_VEGA OR MIOPEN_TEST_GFX908 OR MIOPEN_TEST_HIP_NOGPU))
+    find_program(ROCMINFO
+        NAMES rocminfo
+        PATHS
+            /opt/rocm
+            ${CMAKE_INSTALL_PREFIX}
+        PATH_SUFFIXES
+            /bin
+    )
+    message(STATUS "rocminfo utility: ${ROCMINFO}")
+    if(ROCMINFO)
+        execute_process (
+            COMMAND ${ROCMINFO}
+            OUTPUT_VARIABLE ROCMINFO_OUTPUT
+            RESULT_VARIABLE ROCMINFO_EXIT_STATUS
+        )
+        if(NOT ROCMINFO_EXIT_STATUS EQUAL 0)
+            message(FATAL_ERROR "ROCMINFO FAILED, GPU TYPE UNKNOWN. Manually set respective MIOPEN_TEST_GFX* CMake variable to specify target GPU for testing.")
+        endif()
+        if(ROCMINFO_OUTPUT MATCHES "gfx900|gfx906")
+            set(MIOPEN_TEST_VEGA ON)
+        elseif(ROCMINFO_OUTPUT MATCHES "gfx908")
+            set(MIOPEN_TEST_GFX908 ON)
+        else()
+            message(WARNING "TESTING IS NOT SUPPORTED FOR THE DETECTED GPU")
+        endif()
+    else()
+        message(FATAL_ERROR "ROCMINFO NOT FOUND, GPU TYPE UNKNOWN. Manually set respective MIOPEN_TEST_GFX* CMake variable to specify target GPU for testing.")
```

Comment:
We probably dont want to make this a fatal error because this will fail when the user just wants to build MIOpen and not run the tests(perhaps the user builds on a machine with no gpus and then copies the binaries to the machines with GPUs).

If you want an error then you can add a test which prints out the message and then fails:

```cmake
add_test(NAME rocminfo_check COMMAND echo "ROCMINFO FAILED: ...")
set_tests_properties(rocminfo_check PROPERTIES FAIL_REGULAR_EXPRESSION "FAILED")
```

Or you could add a custom target that `tests` depends on and prints out a failed message as well(this will show an error before running the tests):

```cmake
add_custom_target(rocminfo_check COMMAND echo "ROCMINFO FAILED ..." COMMAND exit 1)
add_dependencies(tests rocminfo_check)
```

Reply:
@pfultz2 Good idea, let me try that...

Reply:
@pfultz2 Yep, that's better. Thanks for the idea.